### DISCUSSION TITLE
Hosting nautobot in a /subfolder on kubernetes.

### DISCUSSION THREAD

**Discussion by ssid334** [nautobot/nautobot#5141] [General]
Hey,

I'm trying to build development environment using microk8s. I have no problem using nginx ingres and host the application on the root address of my cluster (so when I enter https://www.my-nautobot.net/ i'm getting the page and everything is working fine.

I'm using version 1.6.2 and the helm charts published on github.

But I would like to host multiple instances on a single cluster. I have created separated namespaces and I installed nautobot there and this part is working fine. Reconfiguring ingress seems to be straightforward too: I point given "subfolder" url to the service in a given namespace.

For example:

/instance1 (https://www.my-nautobot.net/instance1) points to namespace instance1 that is having nautobot installed
/instance2 (https://www.my-nautobot.net/instance2) points to another namespace
and so on.

Now this is the place where I'm strugging with:
1. I tried to use django parameter: `FORCE_SCRIPT_NAME = "/instance1"` or `FORCE_SCRIPT_NAME = "/instance1/"` in naubot_config
2. I tried to add "X-Script-Name: /instance1" on ingress level

and none of the above seems to be working as hoped.

After configuring any of the above, I get 404 response when I enter the url.
What I got working is static files configuration: `STATIC_URL = "/instance1/static/"` <- this affects static files correctly, I'm able to download any javascript library/themes/documentation etc correctly. 

This is the result I'm getting from the UI when accessing given instance.
![image](https://github.com/nautobot/nautobot/assets/29715602/4229623b-643d-4904-b332-4033bc764ed7)

What I noticed is that "log in" link in upper right looks like this:

https://www.my-nautobot.net/instance1/login/?next=/instance1/instance1
So the "instance1" string is duplicated.

Do I miss something? Is there any other parameter that I should be used? Do you have a guideline how to address it?
Is that configuration supported, because maybe it isn't?

Thank you in advance for any comments regarding this.

**Comment by whitej6** [comment]
Are you using the official helm charts for managing this deployment or separate manifests?

**Reply by ssid334** [reply]
that is correct, I'm using officiall helm chart, so far it's 2.0.4 version of the helm chart.

This gave me thinking: should I open this discussion for the nautbot helm chart repo?

**Reply by whitej6** [reply]
It can stay here, let me dig into the chart. I can't recall if it has out of the box support for path based routing or not.
